### PR TITLE
revert setuptools pinning, do not override CC, CXX

### PR DESCRIPTION
**Pull request by mattip**
Try reverting #4510 and also not setting CC/CXX. Let setuptools/distutils set the compilers. Note the ci-run.sh script would set CC/CXX, but the wheel building would not.

### CODE REVIEW

File: Tools/ci-run.sh [hunk 1]
```diff
@@ -22,10 +22,8 @@ elif [[ $OSTYPE == "linux-gnu"* ]]; then
 
   sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$GCC_VERSION 60 $ALTERNATIVE_ARGS
 
-  export CC="gcc"
   if [[ $BACKEND == *"cpp"* ]]; then
     sudo update-alternatives --set g++ /usr/bin/g++-$GCC_VERSION
-    export CXX="g++"
   fi
 elif [[ $OSTYPE == "darwin"* ]]; then
   echo "Setting up macos compiler"
```
Comment (mattip) [inline Tools/ci-run.sh:29]: Maybe remove the overrides below as well, and use CFLAGS to set the options?

Comment (scoder) [reply Tools/ci-run.sh:29]: Yeah, that seems better.

Comment (mattip) [reply Tools/ci-run.sh:29]: Would you like me to do that in this PR (assuming the linux variant passes CI)?

Comment (scoder) [reply Tools/ci-run.sh:29]: Yes, please give it a try here.

**Comment by scoder** [comment]
BTW, you can search the logs for the output of the `buildenv` "test", which just prints debug information about the running system.

**Comment by scoder** [comment]
The distutils settings printed by `buildenv` look the same in all builds. However, `runtests.py` imports distutils first, before it imports setuptools (only as a test dependency). That means that it probably sets up the whole machinery of CPython's distutils before setuptools gets to replace it again. Can't say if that gets in the way here, but it easily could.

**Comment by mattip** [comment]
Yeah, this is going to be a whack-a-mole to find all the places distutils is imported and preemtively import setuptools somewhere earlier in the process. If cython can accept a hard dependency on setuptools, one solution may be to use setuptools everywhere distutils is currently used, i.e. instead of `from distutils.extension import Extension` do `from setuptools import Extension`

**Comment by scoder** [comment]
> If cython can accept a hard dependency on setuptools, one solution may be to use setuptools everywhere distutils is currently used

I generally agree: https://github.com/cython/cython/issues/4405#issuecomment-996785986

With the exception of a case like this:
https://github.com/cython/cython/blob/c6f5c5ddcc021099febae7cb1194fedea01fd056/Cython/Distutils/build_ext.py#L9-L23

**Comment by mattip** [comment]
Closing, out of date

**Pull request closed without merging**